### ISSUE TITLE
Publish 0.5 to crates.io

### ISSUE THREAD

**Issue by pcwalton** [serde-rs/serde#124]
Bincode is blocked on publishing 0.5 to crates.io. Are we ready to do this?

cc @erickt

**Comment by oli-obk** [comment]
* xml decoding is ready for serde 0.5 (tested, working)
* pandoc filters are blocked on serde 0.5 (tested, working)

I don't require any more breaking changes.

cc @hugoduncan @Byron

**Comment by tomprogrammer** [comment]
In `serde::de::Error`:

Maybe we could strip off the `_error` suffix, it's already clear due to the name of the trait. I usually call these method using `de::Error::syntax_error()`.

I often parse some data coming in via `visit_str` and other visitors into specialized types. There are additional error cases when a constraint is not met, for example parsing a string as UUID. How to handle such errors?

One could argue that these errors are not related to serialization/deserialization, but validation. That means that validation errors should be embedded (like it's done in `io::Error`?).

Otherwise we could add some more methods to `de::Error`. I'd like to change `fn syntax_error() -> Self` to `fn syntax(msg: &str) -> Self` because I (ab)use it to signal validation errors and have no way to transport which constraint was violated. Which error conditions you like to see included?

I guess I should file this as a new issue. What do you think, should we postpone this discussion after 0.5.0 to not block new implementations (bincode, xml, etc.) or include it to minimize frequent breakage?

```rust
pub trait Error {
    fn syntax_error() -> Self;

    fn end_of_stream_error() -> Self;

    fn unknown_field_error(field: &str) -> Self;

    fn missing_field_error(field: &'static str) -> Self;
}
```

**Comment by tomprogrammer** [comment]
I opened https://github.com/serde-rs/serde/issues/125, better collect thoughts there instead of filling this issue with unrelated content.

**Comment by erickt** [comment]
Hello everyone. I apologize for the delay, I found a bug in the json float parsing that's taken a little bit to fix. I've just pushed it up as #126.

**Comment by erickt** [comment]
0.5.0 has been published!

**Comment by oli-obk** [comment]
:tada:
